src/crawlers/puppeteer_crawler.js

```diff
@@ -287,6 +287,7 @@ class PuppeteerCrawler {
 
         try {
             const response = await this.gotoFunction({ page, request, autoscaledPool, puppeteerPool: this.puppeteerPool });
+            await this.puppeteerPool.serveLiveViewSnapshot(page);
```

Comment:
Can you run this in parallel with `this.handlePageFunction`?

Reply:
The snapshotting runs in parallel, just its initialization is awaited because getting browser's pages is `async`, as well as finding our `PuppeteerInstance` using an existing `Browser`.